orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryUtilSpec.groovy

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipeline.persistence
+
+import spock.lang.Specification
+
+import static com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner.STAGE_AFTER
+import static com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner.STAGE_BEFORE
+import static com.netflix.spinnaker.orca.test.model.ExecutionBuilder.orchestration
+import static com.netflix.spinnaker.orca.test.model.ExecutionBuilder.stage
+
+class ExecutionRepositoryUtilSpec extends Specification {
+
+  def "should order stages by ref id"() {
+    given:
+    def one = stage {
+      type = "three"
+      refId = "3"
+    }
+    def two = stage {
+      type = "one"
+      refId = "1"
+    }
+    def three = stage {
+      type = "two"
+      refId = "2"
+    }
+    def execution = orchestration {}
+
+    when:
+    ExecutionRepositoryUtil.sortStagesByReference(execution, [three, one, two])
+
+    then:
+    execution.stages*.refId == ["1", "2", "3"]
```

Comment:
Is there any reason why this would actually be deterministic?

Reply:
It turns out that deck expects the stage ordering to be deterministic in order for things to render correctly.